internal/pkg/service/buffer/store/model/mapper/mapper.go

```diff
@@ -0,0 +1,197 @@
+	}, nil
+}
+
+func ReceiverBaseFromPayload(projectID int, payload *buffer.CreateReceiverPayload) *model.ReceiverBase {
+	name := payload.Name
+
+	// Generate receiver ID from Name if needed
+	var id string
+	if payload.ID != nil && len(*payload.ID) != 0 {
+		id = strhelper.NormalizeName(string(*payload.ID))
+	} else {
+		id = strhelper.NormalizeName(name)
+	}
+
+	// Generate Secret
+	secret := idgenerator.ReceiverSecret()
+
+	return &model.ReceiverBase{
+		ID:        id,
+		ProjectID: projectID,
+		Name:      name,
+		Secret:    secret,
+	}
+}
+
+func ExportBaseFromPayload(payload *buffer.CreateExportData) (r *model.ExportBase, err error) {
+	name := payload.Name
+
+	// Generate export ID from Name if needed
+	var id string
+	if payload.ID != nil && len(*payload.ID) != 0 {
+		id = string(*payload.ID)
+	} else {
+		id = strhelper.NormalizeName(name)
+	}
+
+	// Handle conditions with defaults
+	conditions := model.DefaultConditions()
+	if payload.Conditions != nil {
+		conditions.Count = payload.Conditions.Count
+		conditions.Size, err = datasize.ParseString(payload.Conditions.Size)
+		if err != nil {
+			return nil, serviceError.NewBadRequestError(errors.Errorf(
+				`value "%s" is not valid buffer size in bytes. Allowed units: B, kB, MB. For example: "5MB"`,
+				payload.Conditions.Size,
+			))
+		}
+		conditions.Time, err = time.ParseDuration(payload.Conditions.Time)
+		if err != nil {
+			return nil, serviceError.NewBadRequestError(errors.Errorf(
+				`value "%s" is not valid time duration. Allowed units: s, m, h. For example: "30s"`,
+				payload.Conditions.Size,
+			))
+		}
+	}
+
+	return &model.ExportBase{
+		ID:               id,
+		Name:             name,
+		ImportConditions: conditions,
+	}, nil
+}
+
+func MappingFromPayload(payload *buffer.Mapping) (*model.Mapping, error) {
+	// mapping
+	tableID, err := model.ParseTableID(payload.TableID)
+	if err != nil {
+		return nil, err
+	}
+	columns := make([]column.Column, 0, len(payload.Columns))
+	for _, columnData := range payload.Columns {
+		c, err := column.TypeToColumn(columnData.Type)
+		if err != nil {
+			return nil, err
+		}
+		if template, ok := c.(column.Template); ok {
+			if columnData.Template == nil {
+				return nil, serviceError.NewBadRequestError(errors.Errorf("missing template column data"))
+			}
+			template.Language = columnData.Template.Language
+			template.UndefinedValueStrategy = columnData.Template.UndefinedValueStrategy
+			template.DataType = columnData.Template.DataType
+			template.Content = columnData.Template.Content
+			c = template
+		}
+		columns = append(columns, c)
+	}
+
+	return &model.Mapping{
+		TableID:     tableID,
+		Incremental: payload.Incremental == nil || *payload.Incremental, // default true
+		Columns:     columns,
+	}, nil
+}
+
+func formatReceiverURL(bufferAPIHost string, projectID int, receiverID string, secret string) string {
+	return fmt.Sprintf("https://%s/v1/import/%d/%s/%s", bufferAPIHost, projectID, receiverID, secret)
+}
```

Comment:
Tohle je už jednou v servise: https://github.com/keboola/keboola-as-code/blob/7d2b42c3972ba9b33efedc3135562d2f67d6de08/internal/pkg/service/buffer/api/service/service.go#L453-L455 Mohl bys to prosím tě sjednotit, tj. nechat jen na jednom místě?

Reply:
🤔 nemělo by to `v1` být aspoň v nějaký konstantě? (pokud se to nedá vytáhnout odněkud z Goa)

Reply:
> nemělo by to v1 být aspoň v nějaký konstantě?

O tom už jsme se myslím někdy v minulosti bavili a dohodlo se, že tam bude aspoň test. Na ten jsem ale zapomněl, tak jsem ho teď přidal

Reply:
ok 👍